frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/DivergedFromRemoteDialog.kt

```diff
@@ -0,0 +1,104 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.dsl.builder.MutableProperty
+import com.intellij.ui.dsl.builder.panel
+import com.virtuslab.gitmachete.backend.api.IManagedBranchSnapshot
+import com.virtuslab.gitmachete.backend.api.IRemoteTrackingBranchReference
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus
+import com.virtuslab.gitmachete.frontend.actions.compat.buttonsGroupCompat
+import com.virtuslab.gitmachete.frontend.actions.compat.commentCompat
+import com.virtuslab.gitmachete.frontend.actions.compat.rowCompat
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.fmt
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString
+
+enum class DivergeResolutionOption {
+  DO_NOT_SYNC,
+  FORCE_PUSH,
+  RESET_ON_REMOTE
+}
+
+class DivergedFromRemoteDialog(
+  project: Project,
+  private val remoteBranch: IRemoteTrackingBranchReference,
+  private val branch: IManagedBranchSnapshot,
+  private val relationToRemote: SyncToRemoteStatus,
+  private val withResetOption: Boolean
+) : DialogWrapper(project, /* canBeParent */ true) {
+
+  private var divergeResolutionOption = DivergeResolutionOption.DO_NOT_SYNC
+
+  init {
+    title =
+      getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.title")
+    setOKButtonMnemonic('O'.code)
+    setCancelButtonText(getString("action.GitMachete.BaseTraverseAction.dialog.cancel-traverse"))
+    super.init()
+  }
+
+  fun showAndGetThePreferredAction() =
+    if (showAndGet()) {
+      divergeResolutionOption
+    } else {
+      null
+    }
+
+  override fun createCenterPanel() = panel {
+    rowCompat {
+      label(
+        fmt(
+          getString(
+            "action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.text.HTML"
+          ),
+          branch.name,
+          remoteBranch.name,
+          getRelationToRemoteDescription()
+        )
+      )
+    }
+    buttonsGroupCompat {
+      rowCompat {
+        radioButton(
+          getString(
+            "action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.do-not-resolve-option"
+          ),
+          DivergeResolutionOption.DO_NOT_SYNC
+        ).commentCompat(getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.do-not-resolve-option.comment"))
+      }
+
+      rowCompat {
+        radioButton(
+          getString(
+            "action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.force-push-option"
+          ),
+          DivergeResolutionOption.FORCE_PUSH
+        ).commentCompat(getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.force-push.comment"))
+      }
+
+      if (withResetOption) {
+        rowCompat {
```

Comment:
As discussed, TBD if these two options should ever be suggested at the same time

Reply:
And this also poses a question whether radio buttons are really needed (maybe just stick to "regular" buttons at the bottom of the dialog?)